src/zip.js

```diff
@@ -22,28 +26,43 @@ const zipFunctions = async function (
     jsExternalModules = [],
     jsIgnoredModules = [],
     parallelLimit = DEFAULT_PARALLEL_LIMIT,
-    skipGo,
+    skipGo = true,
     zipGo,
   } = {},
 ) {
-  const [srcPaths, pluginsModulesPath] = await Promise.all([getSrcPaths(srcFolder), getPluginsModulesPath(srcFolder)])
+  await makeDir(destFolder)
 
+  const paths = await listFunctionsDirectory(srcFolder)
+  const [functions, pluginsModulesPath] = await Promise.all([
+    runtimes.getFunctionsFromPaths(paths, { dedupe: true }),
+    getPluginsModulesPath(srcFolder),
+  ])
   const zipped = await pMap(
-    srcPaths,
-    (srcPath) =>
-      zipFunction(srcPath, destFolder, {
+    functions.values(),
+    async (func) => {
+      const zipResult = await func.runtime.zipFunction({
+        destFolder,
+        extension: func.extension,
+        filename: func.filename,
         jsBundler,
         jsExternalModules,
         jsIgnoredModules,
+        mainFile: func.mainFile,
         pluginsModulesPath,
-        skipGo,
-        zipGo,
-      }),
+        runtime: func.runtime,
+        srcDir: func.srcDir,
+        srcPath: func.srcPath,
+        stat: func.stat,
+        zipGo: zipGo === undefined ? !skipGo : zipGo,
```

Comment:
Is the new default value for `skipGo` and this line needed (as opposed to forwarding `skipGo` and `zipGo` as is? Considering `zipFunction()` default values already assign those parameters.

Reply:
I've added it because `zipFunctions` is no longer calling the `zipFunction` from this file as it used to. It's now directly calling the `zipFunction` method of the corresponding runtime, where the default for `zipGo` is not defined.

Reply:
Makes sense. :+1: 
We should eventually remove `skipGo` and/or `zipGo`, because the default value is starting to be confusing to guess.

Reply:
Agree!